ccdproc/tests/test_image_collection.py

```diff
@@ -233,12 +233,16 @@ def test_generator_data(self, triage_setup):
         for img in collection.data():
             assert isinstance(img, np.ndarray)
 
-    def test_generator_ccds(self, triage_setup):
+    def test_generator_ccds_without_unit(self, triage_setup):
         collection = image_collection.ImageFileCollection(
                 location=triage_setup.test_dir, keywords=['imagetyp'])
-        for ccd in collection.ccds():
-            assert isinstance(ccd, CCDData)
 
+        with pytest.raises(ValueError):
+            ccd = collection.ccds()[0]
```

Comment:
Even if this test works, it doesn't test what you want because one cannot subscript generators. You need to call `next(collections.ccds())`